### DISCUSSION TITLE
Does maplibre support China crs likes gcj02

### DISCUSSION THREAD

**Discussion by tinyboat9677** [maplibre/maplibre-gl-js#2224] [General]
China map has different crs ,some offset to webmercator, anyone can solve this?

**Comment by zerda-ocm** [comment]
Solve what? You have to be a little bit more specific with your problem.

**Reply by exotfboy** [reply]
I guess he means supporting `EPSG:4326` or `EPSG:4490` crs in MapLibre.

**Comment by wipfli** [comment]
MapLibre only supports one projection at the moment.

**Reply by 1ec5** [reply]
GCJ-02 is a coordinate reference system, not a projection. It’s an obfuscated variant of WGS-84 that the PRC government mandates for all maps. Mobile platforms are required to have their GNSS functionality conform to GCJ-02, so mobile users in China require a basemap that also conforms, or the user location will be way off.

Mapbox used to develop a “[China plugin](https://blog.mapbox.com/mobile-plugins-for-mapbox-cn-d59c7ab5452b)” that implemented this CRS, but I don’t think it’s available anymore. There are open-source libraries to convert between WGS-84 and GCJ-02, based on either leaked state secrets or reverse engineering, but these libraries aren’t affiliated with MapLibre or Mapbox and would need to be obtained separately.

**Reply by wipfli** [reply]
Ah is that the marsian thing?

**Reply by 1ec5** [reply]
Yes.
